src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -69,145 +69,98 @@ internal bool TryGetNextHeader(Stream archiveStream, bool copyData)
-                }
-            }
-            if (gea.TryGetValue(PaxEaMTime, out string? paxEaMTime))
-            {
-                if (TarHelpers.TryConvertToDateTimeOffset(paxEaMTime, out DateTimeOffset mTime))
-                {
-                    _mTime = mTime;
-                }
-            }
-            if (gea.TryGetValue(PaxEaMode, out string? paxEaMode))
-            {
-                _mode = Convert.ToInt32(paxEaMode);
-            }
-            if (gea.TryGetValue(PaxEaUid, out string? paxEaUid))
-            {
-                _uid = Convert.ToInt32(paxEaUid);
-            }
-            if (gea.TryGetValue(PaxEaGid, out string? paxEaGid))
-            {
-                _gid = Convert.ToInt32(paxEaGid);
-            }
-            if (gea.TryGetValue(PaxEaUName, out string? paxEaUName))
-            {
-                _uName = paxEaUName;
-            }
-            if (gea.TryGetValue(PaxEaGName, out string? paxEaGName))
-            {
-                _gName = paxEaGName;
-            }
-        }
-
         // Reads the elements from the passed dictionary, which comes from the previous extended attributes entry,
         // and inserts or replaces those elements into the current header's dictionary.
         // If any of the dictionary entries use the name of a standard attribute, that attribute's value gets replaced with the one from the dictionary.
         // Unlike the historic header, numeric values in extended attributes are stored using decimal, not octal.
         // Throws if any conversion from string to the expected data type fails.
         internal void ReplaceNormalAttributesWithExtended(IEnumerable<KeyValuePair<string, string>> extendedAttributesEnumerable)
         {
-            Dictionary<string, string> ea = new Dictionary<string, string>(extendedAttributesEnumerable);
-            if (ea.Count == 0)
+            Debug.Assert(_extendedAttributes == null);
+            _extendedAttributes = new Dictionary<string, string>(extendedAttributesEnumerable);
+            if (_extendedAttributes.Count == 0)
             {
                 return;
             }
-            _extendedAttributes ??= new Dictionary<string, string>();
 
-            // First step: Insert or replace all the elements in the passed dictionary into the current header's dictionary.
-            foreach ((string key, string value) in ea)
-            {
-                _extendedAttributes[key] = value;
-            }
+            // Find all the extended attributes with known names and save them in the expected standard attribute.
 
-            // Second, find all the extended attributes with known names and save them in the expected standard attribute.
-            if (ea.TryGetValue(PaxEaName, out string? paxEaName))
+            // The 'name' header field only fits 100 bytes
+            if (_extendedAttributes.TryGetValue(PaxEaName, out string? paxEaName))
             {
                 _name = paxEaName;
             }
```

Comment:
I don't understand what this comment is telling me.  I don't think TryGetValue is clipping the answer, since it's a standard `Dictionary<T1,T2>`.  Should the body of the if have some trimming in it?  If the value is less than 100 bytes should it be removed from the dictionary because the "normal" attribute is sufficient?

If the comment is saying "While we're copying the PaxEaName value into _name, it'll get trimmed down to 100 UTF-8 bytes when writing, so we don't remove it from the dictionary" then perhaps something more like "Copy an extended name into the header, even though it might get trimmed later".  (Neither the 100, nor the bytes, are relevant to a person reading this line of code... unless there's code nearby that works in those numbers/units)

Reply:
I changed some of the wording around these comments.